app/uk/gov/hmrc/exports/models/declaration/notifications/Notification.scala

```diff
@@ -22,30 +22,55 @@ import play.api.libs.functional.syntax._
 import play.api.libs.json.{Json, Reads, _}
 import uk.gov.hmrc.exports.models.declaration.submissions.SubmissionStatus.SubmissionStatus
 
-case class Notification(
-  actionId: String,
-  mrn: String,
-  dateTimeIssued: ZonedDateTime,
-  status: SubmissionStatus,
-  errors: Seq[NotificationError],
-  payload: String
-)
+case class NotificationDetails(mrn: String, dateTimeIssued: ZonedDateTime, status: SubmissionStatus, errors: Seq[NotificationError])
 
-object Notification {
+object NotificationDetails {
   implicit val readLocalDateTimeFromString: Reads[ZonedDateTime] = implicitly[Reads[LocalDateTime]]
     .map(ZonedDateTime.of(_, ZoneId.of("UTC")))
 
-  implicit val writes: OWrites[Notification] = Json.writes[Notification]
-  implicit val reads: Reads[Notification] =
-    ((__ \ "actionId").read[String] and
-      (__ \ "mrn").read[String] and
+  implicit val writes: OWrites[NotificationDetails] = Json.writes[NotificationDetails]
+  implicit val reads: Reads[NotificationDetails] =
+    ((__ \ "mrn").read[String] and
       ((__ \ "dateTimeIssued").read[ZonedDateTime] or (__ \ "dateTimeIssued").read[ZonedDateTime](readLocalDateTimeFromString)) and
       (__ \ "status").read[SubmissionStatus] and
-      (__ \ "errors").read[Seq[NotificationError]] and
-      (__ \ "payload").read[String])(Notification.apply _)
+      (__ \ "errors").read[Seq[NotificationError]])(NotificationDetails.apply _)
+
+  implicit val format: Format[NotificationDetails] = Format(reads, writes)
+}
+
+case class Notification(actionId: String, payload: String, details: Option[NotificationDetails])
+
+object Notification {
+
+  object DbFormat {
+    implicit val writes: Writes[Notification] =
+      ((JsPath \ "actionId").write[String] and
+        (JsPath \ "payload").write[String] and
+        (JsPath \ "details").writeNullable[NotificationDetails](NotificationDetails.writes))(unlift(Notification.unapply))
+
+    implicit val reads: Reads[Notification] =
+      ((__ \ "actionId").read[String] and
+        (__ \ "payload").read[String] and
+        (__ \ "details").readNullable[NotificationDetails])(Notification.apply _)
+
+    implicit val notificationsReads: Reads[Seq[Notification]] = Reads.seq(reads)
+    implicit val notificationsWrites: Writes[Seq[Notification]] = Writes.seq(writes)
+
+    implicit val format: Format[Notification] = Format(reads, writes)
+  }
 
-  implicit val notificationsReads: Reads[Seq[Notification]] = Reads.seq(reads)
-  implicit val notificationsWrites: Writes[Seq[Notification]] = Writes.seq(writes)
+  object FrontendFormat {
+    implicit val writes: Writes[Notification] = new Writes[Notification] {
+      def writes(notification: Notification) = Json.obj(
+        "actionId" -> notification.actionId,
+        "mrn" -> notification.details.get.mrn,
+        "dateTimeIssued" -> notification.details.get.dateTimeIssued,
+        "status" -> notification.details.get.status,
+        "errors" -> notification.details.get.errors,
```

Comment:
@glidester Just realized that you're using `get` on `Option` here. I'm concerned a bit about it. I know there is a filter in controller not to send Notifications without `details`, but it is still error prone.

Reply:
Fixed